FFCDemoPaymentService/Scheduling/IScheduling.cs

```diff
@@ -0,0 +1,32 @@
+using System;
```

Comment:
I think you're on the right lines with your approach but I think that there are some tweaks you can make.  To keep consistency with language used in this service I'd call this `IScheduleService` and you only should declare signatures in an interface as it just represents a contract.  If you want a concrete implementation a base class is probably better.

In it you just need to declare one signature.  `List<Schedule> CreateSchedule(string claimId, DateTime startDate);

If you want to include schedule length that's fine too.

Then create a `ScheduleService.cs` class to implement the interface which loops and creates a new `Schedule` class.  There's no need for a `ScheduleItem`.  

However, this does highlight an issue in the model I created.  The field that is called `Decimal` is wrong.  It should not be 
[Column("value")]
`public decimal Decimal { get; set; }` 

it should be:

[Column("paymentDate")]
`public DateTime PaymentDate { get; set; }` 

If you can fix that and create a new migration as part of this PR it will make your implementation much easier.